torch/csrc/jit/script/compiler.cpp

```diff
@@ -2391,6 +2392,7 @@ struct to_ir {
       body_block->registerOutput(fn_simple_output);
       node_output = fork_node->output()->setType(
           FutureType::create(fn_simple_output->type()));
+      Inline(body_block);
```

Comment:
Why? runCleanupPasses will run on the subgraph below and do the inlining.

Reply:
`lambdaLiftFork` does sort of DCE i.e. it only lifts values that are used in a subgraph. If we leave a function call in place we end up passing in more values than really necessary and when we recompile the same function in `emitModuleHook` which will contain an already inlined subgraph we end up with different graphs. It's actually a very common theme with all these test failures. We *can't* do inlining early enough to not hinder some of optimizations so we have to re-do them.

Reply:
i.e. running it on a lifted subgraph is too late.